routing/route_weight.hpp

```diff
@@ -84,7 +100,13 @@ class RouteWeight final
   // Regular weight (seconds).
   double m_weight = 0.0;
   // Number of pass-through/non-pass-through area border cross.
-  int m_nonPassThroughCross = 0;
+  // int8_t is enough because we check path length while building the route and do not
+  // allow m_nonPassThroughCross to be too big.
+  int8_t m_nonPassThroughCross = 0;
+  // Number of access=yes/access={private,destination} area border cross.
+  // int8_t is enough because we check path length while building the route and do not
+  // allow m_accessCross to be too big.
+  int8_t m_accessCross = 0;
```

Comment:
Which maximum value could have |m_nonPassThroughCross| and |m_accessCross|?

Reply:
Now it is 2 for m_nonPassThroughCross and `numeric_limits<int8_t>::max() / 2` for m_accessCross (see IndexGraphStarter::CheckLength()).

Furthermore we optimise m_nonPassThroughCross and m_accessCross to have minimal possible values while route building and I don't believe there are some points A and B on Earth such that minimal number of non-pass-through or access cross you need to reach B from A will overflow int8_t. But we still have restriction in IndexGraphStarter::CheckLength() to ensure it will not happen and we'll just have NoRoute in such case.

Reply:
Are you sure? What happens when two waves meet, and in both waves number of access crosses is max / 2?

Reply:
I think yes cause we check length of path including new edge before. 
```C++
          // edge weight included here
503       auto const fullLength = weight + stateV.distance + cur->pS - pV;
504       if (!checkLengthCallback(fullLength))
505         continue;
```
but may be I should decrease this value to make it more robust to AStar implementation details. I think for real world routes we'll never have more than 10 here (bad data where some features of restricted access area are tagged with access and some are not or multiple lift gates) and we'll almost never have more than 2.
WDYT about change to `numeric_limits<int8_t>::max() / 3` (two waves fronts + edge between them for case we'll change AStar implementation)?